### PR TITLE
Refactor panels

### PR DESCRIPTION
**Pull request by scarf005**
#### Summary

SUMMARY: Infrastructure "Refactor Panels"

#### Purpose of change

There's lots of duplicate codes, i mean *a lot*

#### Describe the solution

extract functions

#### Describe alternatives you've considered
port refactored code from DDA?

#### Additional context

- it's still a fraction of what needs to be refactored
- might need C++17 and C++20 features to make it more sane

### CODE REVIEW

File: src/panels_utility.cpp [hunk 1]
```diff
@@ -11,8 +11,6 @@ std::string trunc_ellipse( const std::string &input, unsigned int trunc )
     return input;
 }
 
-// value:   <-  v  ->
-// base : red white green
```
Comment (Zireael07) [inline src/panels_utility.cpp]: I'd keep the comment.

File: src/color.cpp [hunk 2]
```diff
@@ -676,11 +676,11 @@ std::string colorize( const translation &text, const nc_color &color )
     return colorize( text.translated(), color );
 }
 
-std::string get_note_string_from_color( const nc_color &color )
+auto get_note_string_from_color( const nc_color &color ) -> std::string
```
Comment (Coolthulhu) [inline src/color.cpp:679]: We didn't agree on trailing return.
If you get people "on your side" and show that there is support for trailing return in the dev community, I'll OK it. But for now, it's unnecessarily weird.

Comment (scarf005) [reply src/color.cpp:679]: alright, please wait a moment

**Comment by Coolthulhu** [comment]
More lines removed than lines added is always a good sign.

**Pull request merged**